Update EOF validation test for top-level containers

chfast opened the pull request:
- Update tests because of the validation rule change: top-level
  containers must not have truncated data.
- Fix some expected error messages.
- Change ">=Prague" to "Prague" because not supported in retesteth.

chfast committed:
Update EOF validation test for top-level containers

- Update tests because of the validation rule change: top-level
  containers must not have truncated data.
- Fix some expected error messages.
- Change ">=Prague" to "Prague" because not supported in retesteth.

winsvega commented on src/EOFTestsFiller/efExample/validInvalidFiller.yml:
```diff
@@ -92,6 +92,8 @@ validInvalid:
               # Data segment
            0bad  # 2 bytes instead of four
+    expectException:
+      "Prague" : "err: toplevel_container_truncated"
```
you can use pyspec exception style now
EOFException.TOPLEVEL_CONTAINER_TRUNCATED

winsvega commented on src/EOFTestsFiller/efExample/validInvalidFiller.yml:
```diff
@@ -116,7 +118,7 @@ validInvalid:
            0bad60A70BAD  # 6 bytes instead of four
     expectException:
-      ">=Prague" : "EOF_InvalidSectionBodiesSize"
+      "Prague" : "EOF_InvalidSectionBodiesSize"
```
but legacy exceptions are also fine. we are planning to convert json tests into .py eventually anyway

winsvega commented on src/EOFTestsFiller/efValidation/EOF1_truncated_section_Copier.json:
```diff
@@ -32,18 +32,20 @@
         "results": {
           "Prague": {
-            "result": true
+            "exception": "err: toplevel_container_truncated",
```
please use the error code. EOFException.TOPLEVEL_CONTAINER_TRUNCATED  or  retetesteth style

winsvega commented:
```
Test Case "efValidation": (3 of 9)
Error: Test /home/travis/build/ethereum/tests/EOFTests/efValidation/EOF1_undefined_opcodes_.json is outdated. Filler hash is different! ('1c79' != '1d40')  (EOFTests/efValidation/EOF1_undefined_opcodes_Copier, step: CheckFiller)
Error: Tests are not generated (forgot --filltests?): 
 /home/travis/build/ethereum/tests/src/EOFTestsFiller/efValidation/EOF1_undefined_opcodes_Copier.json => /home/travis/build/ethereum/tests/EOFTests/efValidation/EOF1_undefined_opcodes_.json
 ```

chfast committed:
Remove migrated EOF tests

- EOFTests/EIP3540/validInvalid.json: https://github.com/ethereum/execution-spec-tests/pull/598
- EOFTests/efValidation/EOF1_truncated_section_.json: https://github.com/ethereum/execution-spec-tests/pull/740
- EOFTests/efValidation/EOF1_eofcreate_valid_.json: https://github.com/ethereum/execution-spec-tests/pull/738
